Assets/Scripts/Enemy.cs

```diff
@@ -32,4 +39,22 @@ void Update()
         acceleration *= time_step * acceleration_rate / acceleration.magnitude;
         myRigidbody.velocity = Vector2.ClampMagnitude(myRigidbody.velocity + acceleration, max_speed);
     }
+
+    public void TakeDamage(int damage) {
+        curr_health -= damage;
+
+        if(curr_health <= 0){
+            Destroy(gameObject);
```

Comment:
Perhaps we should adopt a policy of creating "early hooks". For example, here we create a function `OnDeath` in `Enemy` which we _currently_ only populate with `Destroy(gameObject)` and leave open later for things like animations or other cleanup.

Reply:
That's a good idea! I'll implement that and push a change in a bit.